src/pudl/settings.py

```diff
@@ -176,6 +176,17 @@ class PhmsaGasSettings(GenericDatasetSettings):
     """The list of years to validate."""
 
 
+class Sec10kSettings(GenericDatasetSettings):
+    """An immutable pydantic model to validate PHMSA settings."""
+
+    data_source: ClassVar[DataSource] = DataSource.from_id("sec10k")
+
+    years: list[int] = range(1993, 2025)
```

Comment:
Doesn't the data only go through 2023 right now?

```suggestion
    years: list[int] = range(1993, 2024)
```

Reply:
Oh yeah we haven't archived the 2024 data yet.